services/galley/src/Galley/API/Teams.hs

```diff
@@ -342,6 +344,25 @@ getTeamMembers zusr tid maxResults = do
       let withPerms = (m `canSeePermsOf`)
       pure (newTeamMemberList mems hasMore, withPerms)
 
+bulkGetTeamMembersH :: UserId ::: TeamId ::: Range 1 HardTruncationLimit Int32 ::: JsonRequest UserIdList ::: JSON -> Galley Response
+bulkGetTeamMembersH (zusr ::: tid ::: maxResults ::: body ::: _) = do
+  UserIdList uids <- fromJsonBody body
+  (memberList, withPerms) <- bulkGetTeamMembers zusr tid maxResults uids
+  pure . json $ teamMemberListJson withPerms memberList
+
+-- | like 'getTeamMembers', but with an explicit list of users we are to return.
+bulkGetTeamMembers :: UserId -> TeamId -> Range 1 HardTruncationLimit Int32 -> [UserId] -> Galley (TeamMemberList, TeamMember -> Bool)
+bulkGetTeamMembers zusr tid maxResults uids = do
+  unless (length uids <= fromIntegral (fromRange maxResults)) $
+    throwM bulkGetMemberLimitExceeded
+  mems <- Data.teamMembersLimited tid uids
+  Data.teamMember tid zusr >>= \case
+    Nothing -> throwM notATeamMember
```

Comment:
Also applies to the existing `getTeamMembers` above, but why do we fetch the result before checking that `zusr` is a member of the team? Should the query not be moved into the `Just m` branch?

Reply:
I agree.  26a4bb6530e4a2dedea21144e012f0ee1bb51ac2